### DISCUSSION TITLE
q-table multiple formats for same column

### DISCUSSION THREAD

**Discussion by RyeBot33** [quasarframework/quasar#15268] [General - Components / Directives / etc] [answered]
Hello,

My issue stems from needing to provide different formats for the same column. For example, I have a column "value" which comes back needing multiple units attached, such as %, °F, °C, etc... per row. Is there a good way to attach these units to the different rows easily? I see there is a row parameter passed into the format function but haven't seen that used in any documentation. My thought was attach a custom function to each row for the units display, but not sure how to accomplish that. Thank you for your time.

**Comment by metalsadman** [comment, marked as answer]
put what have you tried in a codepen. `format` is a function so yeah you can feed a custom one that returns those units based on a condition.

**Reply by RyeBot33** [reply]
Well in writing the code pen I figured it out. I was able to add a property to the row object called 'units' and use that in the format function as row.units. Thank you! Here is the code pen just in case someone else needs it.

https://codepen.io/TheRyeBot/pen/BaPZJVd?editors=1111
